src/EditorFeatures/Test/CodeLens/AbstractCodeLensTest.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.CodeLens
+{
+    public abstract class AbstractCodeLensTest
+    {
+        protected static async Task RunCountTest(XElement input, int cap = 0)
+        {
+            using (var workspace = await TestWorkspace.CreateAsync(input))
+            {
+                foreach (var annotatedDocument in workspace.Documents.Where(d => d.AnnotatedSpans.Any()))
+                {
+                    var document = workspace.CurrentSolution.GetDocument(annotatedDocument.Id);
+                    var syntaxNode = await document.GetSyntaxRootAsync();
+                    foreach (var annotatedSpan in annotatedDocument.AnnotatedSpans)
+                    {
+                        var isCapped = annotatedSpan.Key.StartsWith("c");
```

Comment:
annotated spans can start with any alphanumeric string.  So i would just have it start with "capped".  That would make the test easier to read.